code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -490,33 +490,77 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 /obj/item/clothing/mask/smokable/cigarette/joint
 	name = "joint"
-	desc = "This probably shouldn't ever show up."
+	desc = "A joint lovingly rolled and crafted with care. Blaze it."
 	icon_state = "joint"
+	max_smoketime = 400
+	smoketime = 400
+	chem_volume = 25
+
+/obj/item/clothing/mask/smokable/cigarette/joint/blunt
+	name = "blunt"
+	desc = "A blunt lovingly rolled and crafted with care. Blaze it."
+	icon_state = "cigar"
 	max_smoketime = 500
 	smoketime = 500
-	nicotine_amt = 0
+	nicotine_amt = 4
+	chem_volume = 45
 
-/obj/item/weapon/rollingpaper
+/obj/item/weapon/reagent_containers/rollingpaper
 	name = "rolling paper"
 	desc = "A small, thin piece of easily flammable paper, commonly used for rolling and smoking various dried plants."
 	description_fluff = "The legalization of certain substances propelled the sale of rolling papers through the roof. Now almost every Trans-stellar produces a variety, often of questionable quality."
 	icon = 'icons/obj/cigarettes.dmi'
 	icon_state = "cig paper"
+	volume = 25
+
+/obj/item/weapon/reagent_containers/rollingpaper/blunt
+	name = "blunt wrap"
+	desc = "A small piece of easily flammable paper similar to that which encases cigars. It's made out of tobacco, bigger than a standard rolling paper, and will last longer."
+	icon_state = "blunt paper"
+	volume = 45
 
-/obj/item/weapon/rollingpaper/attackby(obj/item/weapon/W as obj, mob/user as mob)
+/obj/item/weapon/reagent_containers/rollingpaper/attackby(obj/item/weapon/W as obj, mob/user as mob)
 	if (istype(W, /obj/item/weapon/reagent_containers/food/snacks))
 		var/obj/item/weapon/reagent_containers/food/snacks/grown/G = W
-		if (!G.dry)
-			to_chat(user, "<span class='notice'>[G] must be dried before you roll it into [src].</span>")
+		if (!G.dry)                                                                                          //This prevents people from just stuffing cheeseburgers into their joint
+			to_chat(user, "<span class='notice'>[G.name] must be dried before you add it to [src].</span>")
 			return
-		var/obj/item/clothing/mask/smokable/cigarette/joint/J = new /obj/item/clothing/mask/smokable/cigarette/joint(user.loc)
-		to_chat(usr, "<span class='notice'>You roll the [G.name] into a joint!</span>")
-		J.add_fingerprint(user)
+		if (G.reagents.total_volume + src.reagents.total_volume > src.reagents.maximum_volume)               //Check that we don't have too much already in the paper before adding things
+			to_chat(user, "<span class='warning'>The [src] is too full to add [G.name].</span>")
+			return
+		if (src.reagents.total_volume == 0)
+			if (istype(src, /obj/item/weapon/reagent_containers/rollingpaper/blunt))                         //update the icon if this is the first thing we're adding to the paper
+				src.icon_state = "blunt_full"
+			else
+				src.icon_state = "paper_full"
+		to_chat(user, "<span class='notice'>You add the [G.name] to the [src.name].</span>")

```

Comment:
A good pattern to use for stuff like this is `if(!user.unEquip(G, src)) return` - it will move the item into `src` and return `TRUE` if the mob can unequip the item and it successfully moves. Saves having to manually call `drop_from_inventory` and prevents stuff like borg modules ending up in toilets.